sections/producer.tex

```diff
@@ -1,5 +1,26 @@
 \section{Writing a Producer}\label{sec:producer}
 
-The \CCSPStlye{Producer} class is used to calculate higher-level observables and define new columns to be written o disk. The corresponding task is called \CCSPStlye{cf.ProduceColumns} (see Ref.~\cite{cf_repo} for detailed info). Naturally, we only want to compute these new variables for the relevant events for our analysis. Thus, the producers are executed after the selection step. 
+The \CCSPStlye{Producer} class is used to calculate higher-level variables and define new array columns to be written o disk. The corresponding task is called \CCSPStlye{cf.ProduceColumns} (see Ref.~\cite{cf_repo} for detailed info). Naturally, we only want to compute these new variables for the relevant events for our analysis. Thus, the producers are executed after the selection step in the task graph. 
 
-In this part of the tutorial, we will write a producer which calculates the four lepton invariant mass.
+ColumnFlow also provides \CCSPStlye{Producer}s to compute commonly used event information, such as MC, pdf or pileup weights. These can be found under \code{columnflow.production.cms}. The \code{H4L} analysis includes three exemplary \CCSPStlye{Producer}s in \code{h4l/production/example.py}. You will notice that the script starts by importing all relevant modules, including CMS specific ones. We also load both \code{numpy} and \code{awkward} with the \code{maybe\_import} mechanism.
+
+We start by defining a new \CCSPStlye{Producer} class named \code{features}. This class requires the transverse jet momentum \code{Jet.pt}, which must be added to its \code{uses} set, and produces two new array columns, the total jet transverse momentum \code{ht} and the number of jets in an event \code{n\_jet}, which are both added to its \code{produces} set. Each of these new variables is computed and stored to disk using the \code{set\_ak\_column} function. Note that for the case of \code{n\_jet}, we specified that the column element must be an \code{int} value.
```

Comment:
`set_ak_column` doesn't write anything to disk, it just adds a new column to the array. Suggest to modify the sentence accordingly

```suggestion
We start by defining a new \CCSPStlye{Producer} class named \code{features}.
This class requires the transverse jet momentum \code{Jet.pt}, which must be added to its \code{uses} set.
Additionally, it produces two new array columns, the total jet transverse momentum \code{ht} and the number of jets in an event \code{n\_jet}, which are both added to its \code{produces} set.
Each of these new variables is computed and then added to the \code{events} array with the \code{set\_ak\_column} function.
This is necessary to make these variables available outside of the \CCSP{Producer}, e.g.\ for writing the information to disk.
Note that for the case of \code{n\_jet}, we specified that the column element must be an \code{int} value.
```